### PR TITLE
Add nicla_vision_ros2 to rosdistro

### PR DESCRIPTION
**Pull request by torydebra**
# Please Add This Package to be indexed in the rosdistro.

nicla_vision_ros2

# The source is here:

https://github.com/ADVRHumanoids/nicla_vision_ros2

# Checks
 - [x] All packages have a declared license in the package.xml
 - [x] This repository has a LICENSE file
 - [x] This package is expected to build on the submitted rosdistro

### CODE REVIEW

File: humble/distribution.yaml
Comment (sloretz) [inline humble/distribution.yaml:4777]: Are you only adding a doc and source entry for now? If so please delete the "release" section. `bloom-release` will create it for you when you release.

**Review by sloretz — DISMISSED** [review]

**Comment by sloretz** [comment]
[Holding for Humble sync](https://discourse.ros.org/t/preparing-for-humble-sync-2024-06-07/38086)

**Comment by torydebra** [comment]
thanks @sloretz, yes I still do not have a release repo as I am waiting for its approval (https://github.com/ros2-gbp/ros2-gbp-github-org/issues/537), which requires a rosdistro entry itself. 
I removed the release section for now

File: humble/distribution.yaml
Comment (clalancette) [inline humble/distribution.yaml:4772]: These all need to end in `.git` to pass CI:

```suggestion
      url: https://github.com/ADVRHumanoids/nicla_vision_ros2.git
```

**Review by clalancette — CHANGES_REQUESTED** [review]

File: humble/distribution.yaml
Comment (torydebra) [reply humble/distribution.yaml]: ok, sorry, I added it

**Comment by marcoag** [comment]
[Holding for Humble sync](https://discourse.ros.org/t/preparing-for-humble-sync-2024-06-19/38236)

**Review by clalancette — DISMISSED** [review]
This is still failing CI because it is not in alphabetical order.

**Comment by torydebra** [comment]
Hi, it should be fine now :)

**Review by Yadunund — APPROVED** [review]

**Comment by torydebra** [comment]
Hi, CI is failing because xacro is not found. I do not know why it is not being installed, I have it both in cmake and package:

```
find_package(xacro REQUIRED) #to launch nicla_sim
```
```xml
<exec_depend>xacro</exec_depend>
```
it does not like the ```exec_depend```? Should I put the ```depend``` tag instead?


---
```
 By not providing "Findxacro.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "xacro", but
  CMake did not find one.

  Could not find a package configuration file provided by "xacro" with any of
  the following names:

    xacroConfig.cmake
    xacro-config.cmake

  Add the installation prefix of "xacro" to CMAKE_PREFIX_PATH or set
  "xacro_DIR" to a directory containing one of the above files.  If "xacro"
  provides a separate development package or SDK, be sure it has been
  installed.
```

**Comment by marcoag** [comment]
`CMake` will run at build time. Since `xacro` is required when `cmake` runs you need it to be present at build time, not only execution time. The `<depend>` tag will probably solve this one, don't know if there's a better option for your package but you can check [REP 140](https://ros.org/reps/rep-0140.html) and decide.

**Comment by clalancette** [comment]
That said, you should almost never need to do `find_package(xacro)` in your CMakeLists.txt.  You would only need that if you were *linking* against `xacro`.  To just have it available for launching, having the `exec_depend` in the package.xml is enough.

**Comment by torydebra** [comment]
ok thanks for the explanations, I removed the "exec dependecies" from the cmake which were not necessary for building

**Pull request merged**
